internal/pipeline/step_command_matrix.go

```diff
@@ -176,11 +176,23 @@ type MatrixAdjustments []*MatrixAdjustment
 // MatrixAdjustment models an adjustment - a combination of (possibly new)
 // matrix values, and skip/soft fail configuration.
 type MatrixAdjustment struct {
-	With     MatrixAdjustmentWith `yaml:"with"`
-	Skip     bool                 `yaml:"skip,omitempty"`
-	SoftFail bool                 `yaml:"soft_fail,omitempty"`
+	With MatrixAdjustmentWith `yaml:"with"`
+	Skip any                  `yaml:"skip,omitempty"`
 
-	RemainingFields map[string]any `yaml:",inline"`
+	RemainingFields map[string]any `yaml:",inline"` // NB: soft_fail is in the remaining fields
+}
+
+func (ma *MatrixAdjustment) ShouldSkip() bool {
+	switch s := ma.Skip.(type) {
+	case bool:
+		return s
+
+	case string:
+		return s != ""
```

Comment:
Pretty sure the empty string is not falsy in ruby. So if the backend sees
```yaml
skip: ""
```
it will adjust to skip.

Unless, that is, if the backend is using `.emtpy?` or the like. But I don't think it is:
```ruby
      jobs.reject! do |job|
        if job.matrix >= adjustment.with
          job.soft_fail = adjustment.soft_fail

          # Return the skip value so `reject!` will remove matching elements
          adjustment.skip
        end
      end
```

Reply:
yeah, i think you're right. in ruby, only `nil` and `false` are truthy, so i guess we should just check for those